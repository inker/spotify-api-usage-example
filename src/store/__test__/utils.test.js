import {
  getMutationName,
  getMutation,
  getMutationEntry,
  getMutations,
} from '../utils'

const vals = [
  'foo',
  'Bar',
  'camelCase',
  '123',
  'BAZ',
  'qu_u_x',
]

describe('Store utils', () => {
  it('should work for getMutationName', () => {
    const names = vals.map(getMutationName)
    expect(names).toMatchSnapshot()
  })

  it('should work for getMutation', () => {
    const mutation = getMutation('foo')
    const state = {
      foo: 'bar',
      other: 'value',
    }
    mutation(state, 'baz')
    expect(state).toMatchSnapshot()
  })

  it('should work for getMutationEntry', () => {
    const [mutationName, mutation] = getMutationEntry('foo')
    expect(mutationName).toBe('setFoo')
    const state = {
      foo: 'bar',
      other: 'value',
    }
    mutation(state, 'baz')
    expect(state).toMatchSnapshot()
  })

  it('should work for getMutations', () => {
    const state = {
      foo: 'bar',
      other: 'value',
    }
    const mutations = getMutations(Object.keys(state))
    mutations.setFoo(state, 'baz')
    expect(state).toMatchSnapshot()
  })
})
