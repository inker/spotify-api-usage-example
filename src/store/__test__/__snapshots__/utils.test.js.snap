// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store utils should work for getMutation 1`] = `
Object {
  "foo": "baz",
  "other": "value",
}
`;

exports[`Store utils should work for getMutationEntry 1`] = `
Object {
  "foo": "baz",
  "other": "value",
}
`;

exports[`Store utils should work for getMutationName 1`] = `
Array [
  "setFoo",
  "setBar",
  "setCamelCase",
  "set123",
  "setBAZ",
  "setQu_u_x",
]
`;

exports[`Store utils should work for getMutations 1`] = `
Object {
  "foo": "baz",
  "other": "value",
}
`;
